### ISSUE TITLE
`Lint/UselessMethodDefinition` false positive when `super()` has parens

### ISSUE THREAD

**Issue by joeldrapper** [rubocop/rubocop#11427]
The `Lint/UselessMethodDefinition` cop has a false positive warning when you define a method that calls super with parenthesis.

```ruby
def foo = super()
```

This method is not useless because it stops `foo` from passing its block to the super method.

If you define the method to explicitly expect a block, it's fine.

```ruby
def foo(&block) = super()
```

Perhaps it should auto-correct to this form which is clearer rather than auto-correcting to remove the method entirely.

**Comment by sambostock** [comment]
Unless I'm missing something, it looks like **`super()` does forward the block**, as the following test suite passes:

```ruby
require "minitest/autorun"

class SuperWithParensTest < Minitest::Test
  class Parent;
    def block_forwarded? = block_given?
  end

  def test_super_with_parens_forwards_block
    klass = Class.new(Parent) { def block_forwarded? = super() }
    assert(klass.new.block_forwarded? { "passing a block" })
  end

  def test_super_with_parens_and_nil_does_not_forward_block
    klass = Class.new(Parent) { def block_forwarded? = super(&nil) }
    refute(klass.new.block_forwarded? { "passing a block" })
  end
end
```

(`ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]`; not sure if it differs by version)

**Comment by joeldrapper** [comment]
Oh wow, that is `super` weird.
